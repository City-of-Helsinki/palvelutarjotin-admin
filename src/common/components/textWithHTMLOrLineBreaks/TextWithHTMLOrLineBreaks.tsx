import React from 'react';

import TextWithLineBreaks from '../textWithLineBreaks/TextWithLineBreaks';

interface Props {
  text?: string;
  className?: string;
}

/***
 * Checks the presence of HTML elements in the string.
 * The fact is that every string is HTML,
 * but not every HTML contains element nodes (or HTML-tags).
 */
const containsHTMLTags = (text: string): boolean => {
  return /<\/?[a-z][\s\S]*>/i.test(text);
};

const TextWithHTMLOrLineBreaks: React.FC<Props> = ({ text, className }) => {
  if (!text) return null;
  return containsHTMLTags(text) ? (
    <div
      className={className}
      dangerouslySetInnerHTML={{ __html: text ?? '' }}
    />
  ) : (
    <TextWithLineBreaks text={text} className={className} />
  );
};

export default TextWithHTMLOrLineBreaks;
export { containsHTMLTags };
