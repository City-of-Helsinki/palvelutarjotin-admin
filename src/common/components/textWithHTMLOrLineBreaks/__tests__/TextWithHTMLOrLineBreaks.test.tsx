import { render } from '@testing-library/react';
import * as React from 'react';

import TextWithHTMLOrLineBreaks from '../TextWithHTMLOrLineBreaks';

it.each([
  `Line 1
  Line 2`,
  `<p>Line1</p>
  <p>Line2</p>`,
])('TextWithHTMLOrLineBreaks matches snapshot', (text) => {
  const { container } = render(
    <TextWithHTMLOrLineBreaks
      text={text}
      className={'textWithHTMLOrLineBreaks'}
    />
  );
  // eslint-disable-next-line testing-library/no-node-access
  expect(container.firstChild).toMatchSnapshot();
});
