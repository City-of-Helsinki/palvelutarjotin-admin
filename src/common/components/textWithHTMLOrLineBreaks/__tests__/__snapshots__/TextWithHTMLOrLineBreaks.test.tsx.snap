// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithHTMLOrLineBreaks matches snapshot 1`] = `
<div
  class="textWithHTMLOrLineBreaks"
>
  Line 1
  <br />
    Line 2
</div>
`;

exports[`TextWithHTMLOrLineBreaks matches snapshot 2`] = `
<div
  class="textWithHTMLOrLineBreaks"
>
  <p>
    Line1
  </p>
  
  
  <p>
    Line2
  </p>
</div>
`;
